src/extensions/default/JavaScriptRefactoring/main.js

```diff
@@ -66,6 +68,58 @@ define(function (require, exports, module) {
         jsRefactoringEnabled = _isRefactoringEnabled();
     });
 
+    function _handleRefactor(functionName) {
+        var eventName,
+            eventType = "";
+        switch(functionName) {
+        case REFACTOR_RENAME:
+            eventName = REFACTOR_RENAME;
+            eventType = "rename";
+            RenameIdentifier.handleRename();
+            break;
+        case EXTRACTTO_VARIABLE:
+            eventName = EXTRACTTO_VARIABLE;
+            eventType = "extractToVariable";
+            ExtractToVariable.handleExtractToVariable();
+            break;
+        case EXTRACTTO_FUNCTION:
+            eventName = EXTRACTTO_FUNCTION;
+            eventType = "extractToFunction";
+            ExtractToFunction.handleExtractToFunction();
+            break;
+        case REFACTORWRAPINTRYCATCH:
+            eventName = REFACTORWRAPINTRYCATCH;
+            eventType = "tryCatch";
+            WrapSelection.wrapInTryCatch();
+            break;
+        case REFACTORWRAPINCONDITION:
+            eventName = REFACTORWRAPINCONDITION;
+            eventType = "wrapInCondition";
+            WrapSelection.wrapInCondition();
+            break;
+        case REFACTORCONVERTTOARROWFN:
+            eventName = REFACTORCONVERTTOARROWFN;
+            eventType = "convertToFunction";
+            WrapSelection.convertToArrowFunction();
+            break;
+        case REFACTORCREATEGETSET:
+            eventName = REFACTORCREATEGETSET;
+            eventType = "createGetterSetter";
+            WrapSelection.createGettersAndSetters();
+            break;
+        }
+        if (eventName) {
+            // Send analytics data for refactoring
+            HealthLogger.sendAnalyticsData(
```

Comment:
We are not even checking whether we are in js context or not? These commands are available in all type of files. Do we just want capture click on the menu or successful usage of this feature?

Reply:
Added the check for JS context.